### PR TITLE
feat: update version to 3.2

### PR DESCRIPTION
**Pull request by vaeng**

### CODE REVIEW

**Review by glennj — APPROVED** [review]

**Comment by glennj** [comment]
We'll want to coordinate the merge with the dart track PR.

**Comment by Stargator** [comment]
@vaeng I'm curious if this will impact any students that currently have older 2.0 versions of exercises in progress and may result in them failing.

Should the test runner instead support a range of `>=2.0.0 <4.0.0`? To avoid this? Is there a way to track how many students are actively using exercises with Dart 2.0?

**Comment by glennj** [comment]
> I'm curious if this will impact any students that currently have older 2.0 versions of exercises in progress and may result in them failing.

It may, if they are testing locally and submit through the cli.

> Should the test runner instead support a range of `>=2.0.0 <4.0.0`? To avoid this? 

I don't think we can do that. Since dart is pretty slow to get its dependencies, the test runner pre-computes a snapshot that lives in the docker image that runs the tests. We'll be locked into v3.2. If vaeng didn't figure this out, the web editor for dart would be useless.

> Is there a way to track how many students are actively using exercises with Dart 2.0?

I think not. The pharo track, when an exercise is submitted, adds a text file containing details about the development environment, but I think it's the only track that does anything like that.

File: Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ RUN apt-get update && \
 
 WORKDIR /opt/test-runner
 
-COPY pubspec.lock pubspec.yaml ./
+COPY pubspec.yaml ./
```
Comment (ErikSchierboom) [inline Dockerfile:11]: Why was the lock file removed?

Comment (vaeng) [reply Dockerfile:11]: Most likely I did not put it back after I changed something. It is not intentional.

Comment (ErikSchierboom) [reply Dockerfile:11]: Okay, could you revert that?

Comment (vaeng) [reply Dockerfile:11]: As I changed the pubspec I might have to change the lockfile accordingly. With my non-existing dart knowledge, I would take the newly-generated one from the test exercises (where we do not copy the lockfile).
Do you think that is sensible?

Comment (ErikSchierboom) [reply Dockerfile:11]: Agreed!

Comment (vaeng) [reply Dockerfile:12]: Done.

**Review by ErikSchierboom — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
Before we merge this, I think there might be something we need to do first. Ideally, we want to prompt students to upgrade their exercise to the latest version. AFAICT, that is defined in the exercise's `pubspec.yaml` file, right? (see https://github.com/exercism/dart/blob/main/exercises/practice/anagram/pubspec.yaml). If so, we would want to add that file to the `files.invalidator` key of all exercises, which would mean that changes to that file would prompt students to update.

**Comment by glennj** [comment]
> pubspec.yaml ... we would want to add that file to the `files.invalidator` key of all exercises, which would mean that changes to that file would prompt students to update.

I'll add that PR

**Pull request merged**